### DISCUSSION TITLE
how to read cookie value from browser layer ?

### DISCUSSION THREAD

**Discussion by icoco** [eclipse-theia/theia#13840] [Q&A]
as topic,   I try to read some cookie value from browser layer, but face some problems that hard to trace the reasons .

I use js-cookie or other lib,  after struggle, the extension can be build but failed while browser build phase, that seem conflict to application manager layer.

I seek from whole theia code, seem no cookie used in browser layer,   anyone have good solution or face the same problem before ? thanks for your time

**Comment by msujew** [comment]
Hey @icoco,

do you need the cookie only in the browser or also in the backend? We don't use cookies in Theia, because we have no need to transfer information with cookies - all frontend<->backend communication happens via RPC anyway. To store data in the frontend, we use the `localStorage` API of the browser.

**Reply by icoco** [reply]
it is no problem use rpc works on one to one mode, but if one to many, eg, the theia backend as server ,and many browser clients access the server . 

maybe cookie can identify the session or which way can keep the request id ?

**Reply by msujew** [reply]
Can you explain a bit what you're attempting to do? I.e. why you want to use cookies in the first place? It's a bit hard to follow your questions.

**Reply by icoco** [reply]
1) browser layer open address : http://127.0.0.1:300/open?dir=xxx&file=yyy  

2) node layer handle the route path: open, extract the parameters and save it into cache map by current request ID (maybe timestamp: 'ts-177666383' ) , also put the request id  into cookie of the response,  maybe 'requestId'=> timestamp value,  then let response redirect to  http://127.0.0.1:300/#/dir
```
     res.setHeader('Set-Cookie', `requestId=${timestamp}; HttpOnly;Max-Age=60`);
     res.redirect(`/#${dir}`);

```

3) browser layer continue original workflow open the directory , after that we want auto open the file 'yyy' which used in in step 1) : http://127.0.0.1:300/open?dir=xxx&file=yyy
I think that read the 'requestId' from cookie then invoke backend service fetch the parameters values that as store in cache map in step 2.
 
if do not use cookie, what solution can make the workflow works as expectation
